scripts/log-session.py

```
@@ -0,0 +1,334 @@
+def detect_task_id() -> str | None:
+    """Detect task_id from WORKFLOW_STATE.md if available."""
+    try:
+        workflow_state = Path('temp/WORKFLOW_STATE.md')
+        if not workflow_state.exists():
+            return None
+
+        content = workflow_state.read_text()
+        # Look for task reference patterns
+        match = re.search(r'(?:Task ID|TASK)[:\s]+([A-Z]+-\d+)', content, re.IGNORECASE)
+        return match.group(1) if match else None
+    except Exception:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)